### PR TITLE
Backurl fixes

### PR DESCRIPTION
**Pull request by blackboxembedded**
Some fixes for backurl handling support.

Previous discussion/PR attempt: https://github.com/organicmaps/organicmaps/pull/3039

### CODE REVIEW

File: iphone/Maps/Classes/MapViewController.mm [hunk 1]
```diff
@@ -720,10 +720,7 @@ - (void)switchPositionMode {
 
 - (void)goBack {
    NSString *backURL = [DeepLinkHandler.shared getBackUrl];
```
Comment (biodranik) [inline iphone/Maps/Classes/MapViewController.mm:722]: Now this goBack method is called by "Escape" key, and it doesn't check if backURL is empty or valid. Will the app crash if the backurl is missing or is invalid?

I would make a change that returns either nil or valid NSURL from getBackUrl function.

Comment (blackboxembedded) [reply iphone/Maps/Classes/MapViewController.mm:722]: I did a test with no backURL and nothing happened, no crash. 

I will make the suggested changes, thank you for your advice.

Comment (blackboxembedded) [reply iphone/Maps/Classes/MapViewController.mm]: > 

I've made the suggested change, please let me know how it looks. Thank you.

**Review by biodranik — APPROVED** [review]

**Review by biodranik — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**